ReactWindows/ReactNative/ReactInstanceManager.cs

```diff
@@ -0,0 +1,148 @@
+            /// </summary>
+            /// <param name="uiImplementationProvider">The UI Implementaiton provider</param>
+            /// <returns></returns>
+            public Builder SetUIImplementationProvider(UIImplementationProvider uiImplementationProvider)
+            {
+                _UIImplementationProvider = uiImplementationProvider;
+                return this;
+            }
+
+            /// <summary>
+            /// Path to the JS bundle file to be loaded from the file system.
+            /// </summary>
+            /// <param name="jsBundleFile">JS bundle file path</param>
+            /// <returns>A builder instance</returns>
+            public Builder SetJSBundleFile(string jsBundleFile)
+            {
+                _jsBundleFile = jsBundleFile;
+                return this;
+            }
+            
+            /// <summary>
+            /// Path to your app's main module on the packager server. This is used when
+            /// reloading JS during development. All paths are relative to the root folder
+            /// the packager is serving files from.
+            /// </summary>
+            /// <param name="jsMainModuleName"></param>
+            /// <returns></returns>
+            public Builder SetJSMainModuleName(string jsMainModuleName)
+            {
+                _jsMainModuleName = jsMainModuleName;
+                return this;
+            }
+
+            public Builder AddPackage(IReactPackage reactPackage)
+            {
+                _reactPackages.Add(reactPackage);
+                return this;
+            }
+
+            /// <summary>
+            /// Instantiates a new {@link ReactInstanceManagerImpl}.
+            /// </summary>
+            /// <param name="initialLifecycleState"></param>
+            /// <returns></returns>
+            public Builder SetInitialLifecycleState(LifecycleState initialLifecycleState)
+            {
+                _LifecycleState = initialLifecycleState;
+                return this;
+            }
+
+            /// <summary>
+            /// Instantiates a new <see cref="ReactInstanceManagerImpl"/> .
+            /// Before calling <see mref="build"/>, the following must be called: setApplication then setJSMainModuleName
+            /// </summary>
+            /// <returns>A ReactInstanceManager instance</returns>
+            public ReactInstanceManager Build()
+            {
+                if (_jsBundleFile == null || _jsMainModuleName == null)
+                {
+                    throw new ArgumentNullException("JS bundle information is not provided exception");
+                }
+
+                if (_UIImplementationProvider == null)
+                {
+                    // create default UIImplementationProvider if the provided one is null.
+                    _UIImplementationProvider = new UIImplementationProvider();
+                }
+
+                return new ReactInstanceManagerImpl(_jsMainModuleName, _reactPackages,
```

Comment:
put each input parameter on a new line, it makes it so if the parameters change in the future, we have a smaller diff.